Fill.py

```diff
@@ -835,49 +837,35 @@ def swap_location_item(location_1: Location, location_2: Location, check_locked:
-        if force in [True, 'fail', 'failure', 'none', False, 'warn', 'warning']:
-            logging.warning(f'{warning}')
+        if isinstance(force, bool):
+            logging.warning(f"{warning}")
         else:
-            logging.debug(f'{warning}')
+            logging.debug(f"{warning}")
 
     def failed(warning: str, force: typing.Union[bool, str]) -> None:
-        if force in [True, 'fail', 'failure']:
+        if force is True:
             raise Exception(warning)
         else:
             warn(warning, force)
 
-    swept_state = multiworld.state.copy()
-    swept_state.sweep_for_advancements()
-    reachable = frozenset(multiworld.get_reachable_locations(swept_state))
-    early_locations: typing.Dict[int, typing.List[str]] = collections.defaultdict(list)
-    non_early_locations: typing.Dict[int, typing.List[str]] = collections.defaultdict(list)
-    for loc in multiworld.get_unfilled_locations():
-        if loc in reachable:
-            early_locations[loc.player].append(loc.name)
-        else:  # not reachable with swept state
-            non_early_locations[loc.player].append(loc.name)
-
     world_name_lookup = multiworld.world_name_lookup
 
-    block_value = typing.Union[typing.List[str], typing.Dict[str, typing.Any], str]
-    plando_blocks: typing.List[typing.Dict[str, typing.Any]] = []
+    plando_blocks: typing.Dict[int, typing.List[typing.Dict[str, typing.Any]]] = dict()
     player_ids = set(multiworld.player_ids)
     for player in player_ids:
-        for block in multiworld.plando_items[player]:
-            block['player'] = player
-            if 'force' not in block:
-                block['force'] = 'silent'
-            if 'from_pool' not in block:
-                block['from_pool'] = True
-            elif not isinstance(block['from_pool'], bool):
-                from_pool_type = type(block['from_pool'])
-                raise Exception(f'Plando "from_pool" has to be boolean, not {from_pool_type} for player {player}.')
-            if 'world' not in block:
-                target_world = False
-            else:
-                target_world = block['world']
+        plando_blocks[player] = []
+        for block in multiworld.worlds[player].options.plando_items:
+            new_block: typing.Dict[str, typing.Any] = {"player": player}
+            if not isinstance(block.from_pool, bool):
+                raise Exception(f"Plando 'from_pool' has to be boolean, not {type(block.from_pool)} for player {player}.")
+            new_block["from_pool"] = block.from_pool
+            new_block["force"] = block.force
+            target_world = block.world
+
+            if not (isinstance(block.force, bool) or block.force == "silent"):
+                raise Exception(f"Plando `force` has to be boolean or `silent`, not {block.force} for player {player}")
 
             if target_world is False or multiworld.players == 1:  # target own world
```

Comment:
```suggestion
            new_block: typing.Dict[str, typing.Any] = {"player": player}
            new_block["from_pool"] = block.from_pool
            new_block["force"] = block.force
            target_world = block.world
            if target_world is False or multiworld.players == 1:  # target own world
```
If some world dev wanted to do some custom processing of the `PlandoItem` data, they would need to duplicate these checks, and they shouldn't need to.